packages/evalite/src/scorers/faithfulness.ts

```diff
@@ -0,0 +1,205 @@
+import { createScorer } from "../create-scorer.js";
+import { createLLMBasedScorer } from "./base.js";
+import { generateObject } from "ai";
+import { z } from "zod";
+import { failedToScore, isSingleTurnSample } from "./utils.js";
+
+const StatementGeneratorOutputSchema = z.object({
+  statements: z.array(z.string()).describe("The generated statements"),
+});
+
+const StatementFaithfulnessAnswerSchema = z.object({
+  statement: z.string().describe("the original statement, word-by-word"),
+  reason: z.string().describe("the reason of the verdict"),
+  verdict: z
+    .number()
+    .int()
+    .min(0)
+    .max(1)
+    .describe("the verdict (0/1) of the faithfulness"),
+});
+
+const NLIStatementOutputSchema = z.object({
+  statements: z.array(StatementFaithfulnessAnswerSchema),
+});
+
+type NLIStatementOutput = z.infer<typeof NLIStatementOutputSchema>;
```

Comment:
Let's nest all types under the Evalite.Scorers namespace. Helps keep things sane and easy to find for users.